### DISCUSSION TITLE
Spring Sleuth Metrics Not Displaying in Apache SkyWalking UI

### DISCUSSION THREAD

**Discussion by rajeshnayak23** [apache/skywalking#12581] [Q&A]
Dear  Team,

I am experiencing an issue with Apache SkyWalking while monitoring a Java Spring Boot application integrated with skywalking java agent . In the SkyWalking UI, under the "Spring Sleuth" section of the service instance, all metrics display the error: "Metric: [meter_http_server_requests_count] does not exist," resulting in no data being shown.

 The expected behavior is that Spring Sleuth metrics should appear correctly in the SkyWalking UI with all relevant monitoring data.
Can you please provide any insights on why the metric data is showing as "not existing" or how I can ensure that all Spring Sleuth metrics are displayed correctly in Apache SkyWalking?

Best Regards,
Rajesh

**Comment by wu-sheng** [comment]
I feel you don't follow the doc to activate relative metrics

Ref to https://skywalking.apache.org/docs/main/latest/en/setup/backend/micrometer-observations/#dashboard-configuration

**Reply by rajeshnayak23** [reply]
I reviewed the provided documentation on activating the relevant metrics, and it resolved the "Metric: [meter_http_server_requests_count] does not exist" issue. However, I still do not see any data displayed in the "Spring Sleuth" section of the Apache SkyWalking UI.

Could you please let me know if any other configurations are needed or provide guidance on how to ensure that all Spring Sleuth data is displayed correctly in Apache SkyWalking?

**Reply by wu-sheng** [reply]
Do you connect micrometer with the agent by using the toolkit? That is relative.

**Reply by rajeshnayak23** [reply]
I'm currently confused about where to add the toolkit dependencies and make configuration changes for integrating Micrometer with the SkyWalking agent. Should these dependencies be added to the agent's code or within my application's code? Additionally, I'm uncertain whether the required configuration changes should be made on the agent side or the application side. I reviewed the documentation here: [Application Toolkit for Micrometer]   [](https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/application-toolkit-micrometer-1.10/ ) , but I'm still unclear on whether these steps are meant to be applied at the agent level or the application level. Could you please clarify?

**Reply by wu-sheng** [reply]
> Should these dependencies be added to the agent's code or within my application's code?

Your application codes. 

> I'm uncertain whether the required configuration changes should be made on the agent side or the application side.

No agent side change needs to be made. The integration points are through your own application code changes.

**Reply by rajeshnayak23** [reply]
![Screenshot from 2024-08-30 12-20-44](https://github.com/user-attachments/assets/ea93a51f-42ca-435f-b33c-9f3635103f4b)

I have added these configurations in my application and "management.metrics.export.skywalking.enabled=true"  in my application.properties file but still the spring sleuth data is not visible .

java agent version 9.3.0 and the dependecy is 

          <dependency>
    		<groupId>org.apache.skywalking</groupId>
     		<artifactId>apm-toolkit-micrometer-1.10</artifactId>
     		<version>9.3.0</version>
   	</dependency>

**Reply by wu-sheng** [reply]
Then you need to debug agent plugin codes.
Micrometer(redhat) people contributed this plugins. I am not sure the details.

**Reply by rajeshnayak23** [reply]
can you give any other suggestions or ideas how to resolve this issue by mentioning redhat people ?

**Reply by wu-sheng** [reply]
I am not sure which micrometer version you are using. Redhat people contributed plugin with this test scenario to verify this.

https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/resttemplate-6.x-scenario/src/main/java/org/apache/skywalking/testcase/resttemplate/ResttemplateConfiguration.java

**Reply by wu-sheng** [reply]
I never worked for Redhat. @marcingrzejszczak was the one contributed this, but it was 2 year ago.

**Reply by marcingrzejszczak** [reply]
I don't understand what the problem is and what Spring cloud sleuth (feature complete project) has to do with it?

**Reply by rajeshnayak23** [reply]
Hy @marcingrzejszczak  ,
  Actually , while monitoring a Java Spring Boot application integrated with skywalking java agent , the metircs under the service instance Spring Sleuth shows no data. 
How can i get the spring sleuth metirc data?
